src/types/UiaTextRangeBase.cpp

```diff
@@ -1450,6 +1504,178 @@ std::pair<Endpoint, Endpoint> UiaTextRangeBase::_moveByCharacterBackward(gsl::no
     return std::make_pair<Endpoint, Endpoint>(std::move(start), std::move(end));
 }
 
+// Routine Description:
+// - calculates new Endpoints if they were to be moved moveCount times
+// by word.
+// Arguments:
+// - moveCount - the number of times to move
+// - moveState - values indicating the state of the console for the
+// move operation
+// - pAmountMoved - the number of times that the return values are "moved"
+// Return Value:
+// - a pair of endpoints of the form <start, end>
+std::pair<Endpoint, Endpoint> UiaTextRangeBase::_moveByWord(gsl::not_null<IUiaData*> pData,
+                                                            const int moveCount,
+                                                            const MoveState moveState,
+                                                            const std::wstring& wordDelimiters,
+                                                            _Out_ gsl::not_null<int*> const pAmountMoved)
+{
+    if (moveState.Direction == MovementDirection::Forward)
+    {
+        return _moveByWordForward(pData, moveCount, moveState, wordDelimiters, pAmountMoved);
+    }
+    else
+    {
+        return _moveByWordBackward(pData, moveCount, moveState, wordDelimiters, pAmountMoved);
+    }
+}
+
+std::pair<Endpoint, Endpoint> UiaTextRangeBase::_moveByWordForward(gsl::not_null<IUiaData*> pData,
+                                                                   const int moveCount,
+                                                                   const MoveState moveState,
+                                                                   const std::wstring& wordDelimiters,
+                                                                   _Out_ gsl::not_null<int*> const pAmountMoved)
+{
+    // STRATEGY:
+    //    - move the "end" Endpoint to the proper place (if need to move multiple times, do it here)
+    //    - find the "start" Endpoint based on that
+
+    *pAmountMoved = 0;
+    const int count = moveCount;
+    ScreenInfoRow currentScreenInfoRow = moveState.EndScreenInfoRow;
+    Column currentColumn = moveState.EndColumn;
+
+    auto& buffer = pData->GetTextBuffer();
+    for (int i = 0; i < abs(count); ++i)
+    {
+        // get the current row's right
+        const ROW& row = buffer.GetRowByOffset(currentScreenInfoRow);
+        const size_t right = row.GetCharRow().MeasureRight();
+
+        // check if we're at the edge of the screen info buffer
+        if (currentScreenInfoRow == moveState.LimitingRow &&
+            gsl::narrow_cast<size_t>(currentColumn) + 1 >= right)

```

Comment:
Can you do the `gsl::narrow_cast<size_t>(currentColumn) + 1` outside the conditionals since you use its value twice? 

Or otherwise structure this to not do that cast and math twice?